### PR TITLE
non_exhaustive structures

### PR DESCRIPTION
**Pull request by burrbull**
Up MSRV to 1.40

### CODE REVIEW

**Comment by burrbull** [comment]
cc @therealprof 
@adamgreig

**Comment by adamgreig** [comment]
Swapping to non_exhaustive LGTM, should this PR wait until #139 so it doesn't also merge the strict stuff?

**Comment by burrbull** [comment]
> Swapping to non_exhaustive LGTM, should this PR wait until #139 so it doesn't also merge the strict stuff?

At your discretion.

**Comment by therealprof** [comment]
I'm also fine in principle, though I'm not quite clear on the motivation to bump the MSRV just to replace manual non-exhaustive implementation.

**Comment by adamgreig** [comment]
At this point I think 1.40 is old enough that it's nice to be able to clean up the code to use non_exhaustive. Especially since this crate tends to be more "tooling" than "end-user" I'm really not too worried about the MSRV going to 1.40.

**Comment by therealprof** [comment]
I'm not worried about the MSRV bump. I'm simply not a big fan of bumping versions just for cleanup...

**Review by adamgreig — DISMISSED** [review]
We need to bump the version we run clippy under too to support anyhow's clippy config, and this helps tidy up the codebase, so on balance I think let's go for it (and merge #139 as part of this PR).

bors merge

**Comment by adamgreig** [comment]
Ah, this line will need updating before bors sees the tests as passing:

https://github.com/rust-embedded/svd/blob/master/.github/bors.toml#L6

**Comment by adamgreig** [comment]
bors cancel

**Review by adamgreig — APPROVED** [review]
bors merge

**Comment by adamgreig** [comment]
Ah, this failure wasn't being reported but explains why we're not seeing any `build` results against the bors commits:

https://github.com/rust-embedded/svd/actions/runs/695451779

In particular `test strict` should be `test-strict` or similar:

https://github.com/rust-embedded/svd/pull/140/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR48

bors cancel

**Comment by burrbull** [comment]
Rebased

**Comment by burrbull** [comment]
What is wrong with `clippy`?

**Comment by adamgreig** [comment]
It's still complaining about [this line in anyhow](https://github.com/dtolnay/anyhow/blob/master/.clippy.toml#L1):

{"reason":"compiler-message","package_id":"anyhow 1.0.40 (registry+https://github.com/rust-lang/crates.io-index)","target":{"kind":["lib"],"crate_types":["lib"],"name":"anyhow","src_path":"/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.40/src/lib.rs","edition":"2018","doctest":true},"message":{"rendered":"error: error reading Clippy's configuration file `/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.40/.clippy.toml`: unknown field `msrv`, expected one of `blacklisted-names`, `cognitive-complexity-threshold`, `cyclomatic-complexity-threshold`, `doc-valid-idents`, `too-many-arguments-threshold`, `type-complexity-threshold`, `single-char-binding-names-threshold`, `too-large-for-stack`, `enum-variant-name-threshold`, `enum-variant-size-threshold`, `verbose-bit-mask-threshold`, `literal-representation-threshold`, `trivial-copy-size-limit`, `too-many-lines-threshold`, `third-party` at line 1 column 1\n\n","children":[],"code":null,"level":"error","message":"error reading Clippy's configuration file `/home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.40/.clippy.toml`: unknown field `msrv`, expected one of `blacklisted-names`, `cognitive-complexity-threshold`, `cyclomatic-complexity-threshold`, `doc-valid-idents`, `too-many-arguments-threshold`, `type-complexity-threshold`, `single-char-binding-names-threshold`, `too-large-for-stack`, `enum-variant-name-threshold`, `enum-variant-size-threshold`, `verbose-bit-mask-threshold`, `literal-representation-threshold`, `trivial-copy-size-limit`, `too-many-lines-threshold`, `third-party` at line 1 column 1","spans":[]}}

but I don't know why, or why it has worked occasionally...

**Comment by burrbull** [comment]
Switched to use last stable clippy

File: .github/bors.toml
Comment (adamgreig) [inline .github/bors.toml:8]:
```suggestion
  "build (stable)",
  "test",
  "test-strict",
  "clippy_check",
```

**Review by adamgreig — CHANGES_REQUESTED** [review]
One last thing, we should probably add test-strict and clippy_check to the bors config so it requires that they pass.

**Review by adamgreig — APPROVED** [review]
bors merge

**Pull request merged**
